constantine/math/constants/banderwagon.nim

```diff
@@ -0,0 +1,69 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import
+  # Internals
+  ../../platforms/abstractions,
+  ../config/curves,
+  ../arithmetic,
+  ../extension_fields,
+  ../elliptic/ec_twistededwards_projective
+
+func `==`*(P, Q: ECP_TwEdwards_Prj[Fp[Banderwagon]]): SecretBool =
+  ## Equality check for points in the Banderwagon Group
+  ## The equals method is different for the quotient group
+  ## 
+  ## Check for the (0,0) point, which is possible
+  ## 
+  ## This is a costly operation
+
+  var lhs{.noInit.}, rhs{.noInit.}: typeof(P).F
+
+  # Check for the zero points
+  result = not(P.x.is_zero() and P.y.is_zero())
+  result = result or not(Q.x.is_zero() and Q.y.is_zero())
+
+  ## Check for the equality of the points
+  ## X1 * Y2 == X2 * Y1
+  lhs.prod(P.x, Q.y)
+  rhs.prod(Q.x, P.y)
+  result = result and lhs == rhs
+
+func map_to_field*(P: ECP_TwEdwards_Prj[Fp[Banderwagon]]): Fp =
+  ## Map a point on the Banderwagon curve to the base field
+  ## 
+  result.prod(P.x, P.y.inv())
```

Comment:
This can be put in https://github.com/mratsim/constantine/blob/master/constantine/hash_to_curve/hash_to_curve.nim

There will be specialized per curve mappings

Reply:
Actually misread, I thought it was to map random bytes to a field.

How is this used? I don't remember seeing a spec/usage for this.

Reply:
I kinda had a very similar doubt regarding this, but I saw this in the notes of both gottfried and kev. So thought I should include this.

But I haven't found any usage for this anywhere till now. What do you suggest we should do?

Reply:
I think we can remove this at the moment, it's very short and can be reintegrated in a place that makes sense if needed.